services/minespace-web/src/components/pages/Incidents/IncidentPage.js

```diff
@@ -187,34 +280,48 @@ export class IncidentPage extends Component {
     return Promise.resolve();
   };
 
-  handleCreateMineIncident = (values) => {
+  handleCreateMineIncident = (values, isDraft) => {
     this.setState({ isLoaded: false });
+    const message = isDraft ? "Successfully created a draft incident." : null;
     return this.props
-      .createMineIncident(this.props.match.params?.mineGuid, values)
-      .then(({ data: { mine_guid, mine_incident_guid } }) =>
-        this.props.history.replace({
-          pathname: routes.EDIT_MINE_INCIDENT.dynamicRoute(mine_guid, mine_incident_guid),
-          state: { current: 1 },
-        })
-      );
+      .createMineIncident(this.props.match.params?.mineGuid, values, message)
+      .then((response) => {
+        return response?.data;
+      });
   };
 
-  handleUpdateMineIncident = (values) => {
+  handleUpdateMineIncident = (values, isDraft) => {
     const { mineGuid, mineIncidentGuid } = this.props.match.params;
     this.setState({ isLoaded: false });
-    return this.props
-      .updateMineIncident(mineGuid, mineIncidentGuid, values)
-      .then(() => this.handleFetchData())
-      .then(() => this.setState({ isLoaded: true }));
+
+    let message;
+    if (isDraft) {
+      message = "Successfully updated draft incident.";
+    } else if (this.props.location.state.current === 2) {
+      message = "Submitted successfully a new incident.";
```

Comment:
Should this copy read "Successfully submitted a new incident."? To be consistent with the previous message structure.